device/mock_device/src/mock_web_socket_server.rs

```diff
@@ -106,20 +107,78 @@ type WSConnection = Arc<Mutex<HashMap<String, SplitSink<WebSocketStream<TcpStrea
 #[derive(Debug)]
 pub struct MockWebSocketServer {
     mock_data_v2: Arc<RwLock<MockData>>,
-
     listener: TcpListener,
-
     conn_path: String,
-
     conn_headers: HeaderMap,
-
     conn_query_params: HashMap<String, String>,
-
     port: u16,
-
     connected_peer_sinks: WSConnection,
-
     config: MockConfig,
+    /*
+    track thunder methods called and their count per method
+    */
+    stats_channel: ripple_sdk::tokio::sync::mpsc::Sender<String>,
+}
+pub struct StatsCollector {
+    thunder_histogram: Arc<RwLock<HashMap<String, u32>>>,
+    messages: ripple_sdk::tokio::sync::mpsc::Receiver<String>,
+    stats_file: String,
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct ApiStat {
+    method: String,
+    count: u32,
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct ApiStats {
+    stats: Vec<ApiStat>,
+    total: u32,
+}
+impl StatsCollector {
```

Comment:
This is little easter egg: since we have a control point at the thunder level in mock_device, seemed like it was useful/interesting to collect some stats - this will create a histogram if all thunder calls that are serviced by mock_device. The histogram is put in `stats.json` in the eos-ripple directory when (and only when) mock_device is used